graphql/src/schema/api.rs

```diff
@@ -176,6 +167,77 @@ fn add_order_by_type(
     Ok(())
 }
 
+/// Generates enum values for the given set of fields.
+fn field_enum_values(
+    schema: &Document,
+    fields: &[Field],
+) -> Result<Vec<EnumValue>, APISchemaError> {
+    let mut enum_values = vec![];
+    for field in fields {
+        enum_values.push(EnumValue {
+            position: Pos::default(),
+            description: None,
+            name: field.name.to_owned(),
+            directives: vec![],
+        });
+        enum_values.extend(field_enum_values_from_child_entity(
+            schema,
+            field,
+            &field.field_type,
+        )?);
+    }
+    Ok(enum_values)
+}
+
+fn enum_value_from_child_entity_field(
+    schema: &Document,
+    parent_field: &Field,
+    field: &Field,
+) -> Option<EnumValue> {
+    if ast::is_list_or_non_null_list_field(field) || ast::is_entity_type(schema, &field.field_type)
+    {
+        // Sorting on lists or entities is not supported.
+        None
+    } else {
+        Some(EnumValue {
+            position: Pos::default(),
+            description: None,
+            name: format!("{}__{}", parent_field.name, field.name),
+            directives: vec![],
+        })
+    }
+}
+
+fn field_enum_values_from_child_entity(
+    schema: &Document,
+    field: &Field,
+    field_type: &Type,
+) -> Result<Vec<EnumValue>, APISchemaError> {
+    match field_type {
+        Type::NamedType(ref name) => {
+            let named_type = schema
+                .get_named_type(name)
+                .ok_or_else(|| APISchemaError::TypeNotFound(name.clone()))?;
+            Ok(match named_type {
+                TypeDefinition::Object(ot) => ot
+                    .fields
+                    .iter()
+                    .filter_map(|f| enum_value_from_child_entity_field(schema, field, f))
+                    .collect(),
+                TypeDefinition::Interface(i) => i
+                    .fields
+                    .iter()
+                    .filter_map(|f| enum_value_from_child_entity_field(schema, field, f))
+                    .collect(),
+                _ => vec![],
+            })
+        }
+        // Sort on lists is not supported.
+        Type::ListType(ref _t) => Ok(vec![]),
```

Comment:
This should return an error; we assume that this is never called with a list type, if it is there's a programming error somewhere and we should call attention to it.

Reply:
Are you sure? I think it's fine, only the GraphQL Schema is generated here, it's not related to runtime, or GraphQL request.
Returning an empty vector means there are no enum values, which makes sense.